src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -1827,10 +1837,13 @@ public static IntPtr LoadLibrary(string libraryName, Assembly assembly, DllImpor
         /// <exception cref="System.ArgumentNullException">If assembly is null and dllImportSearchPath includes AssemblyDirectory</exception>
         public static bool TryLoadLibrary(string libraryName, Assembly assembly, DllImportSearchPath? searchPath, out IntPtr handle)
         {
+            if (libraryName == null)
+                throw new ArgumentNullException(nameof(libraryName));
+            if (assembly == null)
+                throw new ArgumentNullException(nameof(assembly));
+            
             RuntimeAssembly runtimeAssembly = (assembly != null) ? ((RuntimeAssembly)assembly).GetNativeHandle() : null;
```

Comment:
I think this line can be simplified since we now know that `assembly != null` by the time this line executes.